src/Di/Inject.php

```diff
@@ -20,4 +20,12 @@
      * @var bool
      */
     public $optional = false;
```

Comment:
Not sure, that would break compatibility. But I can investigate if it is possible at all